File: apps/app-4/backend/internal/core/domain/waiting_list.go
package domain

import "time"

// WaitingList represents a patient currently in the virtual waiting room.
type WaitingList struct {
	ID        int       `gorm:"primaryKey;autoIncrement"`
	PatientID int       `gorm:"column:patient_id;unique;not null"`
	JoinedAt  time.Time `gorm:"column:joined_at;autoCreateTime"`
	Status    string    `gorm:"column:status;default:waiting"` // waiting, in_consultation, left

	// Relation
	Patient User `gorm:"foreignKey:PatientID;references:ID"`
}

// WaitingListStatus constants
const (
	StatusWaiting        = "waiting"
	StatusInConsultation = "in_consultation"
	StatusLeft           = "left"
)

File: apps/app-4/frontend/src/types/waitingList.ts
import { User } from "./auth";

export interface WaitingListItem {
    id: number;
    patientId: number;
    joinedAt: string;
    status: "waiting" | "in_consultation" | "left";
    patient: User;
}

export interface WaitingRoomState {
    patients: WaitingListItem[];
    status: string; // "connecting", "waiting", "in_consultation", "left"
    lastUpdated: number;
}
